### PR TITLE
Refactor app

### PR DESCRIPTION
**Pull request by leaanthony**
* Refactor the way the main application works as a starting point for #1480 and #1521 
* Clean up `app` package
* Remove need for application options in AssetServer
* Add experimental programmatic API that uses the refactored app

### CODE REVIEW

File: v2/internal/frontend/assetserver/assethandler.go [hunk 1]
```diff
@@ -33,8 +32,7 @@ type assetHandler struct {
 	retryMissingFiles bool
 }
 
-func NewAssetHandler(ctx context.Context, options *options.App) (http.Handler, error) {
-	vfs := options.Assets
+func NewAssetHandler(ctx context.Context, vfs iofs.FS, AssetsHandler http.Handler) (http.Handler, error) {
```
Comment (stffabi) [inline v2/internal/frontend/assetserver/assethandler.go:35]: The assetsHandler parameter should shard with a lower case.
```suggestion
func NewAssetHandler(ctx context.Context, vfs iofs.FS, assetsHandler http.Handler) (http.Handler, error) {
```

File: v2/internal/frontend/assetserver/assethandler.go [hunk 2]
```diff
@@ -53,7 +51,7 @@ func NewAssetHandler(ctx context.Context, options *options.App) (http.Handler, e
 
 	result := &assetHandler{
 		fs:      vfs,
-		handler: options.AssetsHandler,
+		handler: AssetsHandler,
```
Comment (stffabi) [inline v2/internal/frontend/assetserver/assethandler.go:54]:
```suggestion
		handler: assetsHandler,
```

File: v2/internal/frontend/assetserver/assetserver.go [hunk 3]
```diff
@@ -31,8 +31,8 @@ type AssetServer struct {
 	appendSpinnerToBody bool
 }
 
-func NewAssetServer(ctx context.Context, options *options.App, bindingsJSON string) (*AssetServer, error) {
-	handler, err := NewAssetHandler(ctx, options)
+func NewAssetServer(ctx context.Context, vfs iofs.FS, AssetsHandler http.Handler, bindingsJSON string) (*AssetServer, error) {
```
Comment (stffabi) [inline v2/internal/frontend/assetserver/assetserver.go:34]:
```suggestion
func NewAssetServer(ctx context.Context, vfs iofs.FS, assetsHandler http.Handler, bindingsJSON string) (*AssetServer, error) {
```

File: v2/internal/frontend/assetserver/assetserver.go [hunk 4]
```diff
@@ -31,8 +31,8 @@ type AssetServer struct {
...
+	handler, err := NewAssetHandler(ctx, vfs, AssetsHandler)
```
Comment (stffabi) [inline v2/internal/frontend/assetserver/assetserver.go:35]:
```suggestion
	handler, err := NewAssetHandler(ctx, vfs, assetsHandler)
```

**Review by stffabi — APPROVED** [review]
LGTM 🚀 , just a minor comment.

**Comment by leaanthony** [comment]
Nice catch! Thanks 👍

File: v2/pkg/options/options.go [hunk 5]
```diff
@@ -47,9 +47,7 @@ type App struct {
 	AlwaysOnTop       bool
 	// BackgroundColour is the background colour of the window
 	// You can use the options.NewRGB and options.NewRGBA functions to create a new colour
-	BackgroundColour *RGBA
-	// RGBA is deprecated. Please use BackgroundColour
-	RGBA               *RGBA
+	BackgroundColour   *RGBA
```
Comment (antimatter96) [inline v2/pkg/options/options.go:50]: Should we add this as a breaking change in release notes ?

Apps working fine on 2.0.0 break on 2.1.0
@leaanthony

Comment (leaanthony) [reply v2/pkg/options/options.go:50]: Yes, though there was fair warning it was being removed. But ultimately yes, looks like it needs adding. Apologies we missed that and thanks for raising it

Comment (antimatter96) [reply v2/pkg/options/options.go:50]: No worries. Getting 2.1 to work with just a 1 line change is amazing.

**Pull request merged**